test/services/database.unit.test.js

```diff
@@ -0,0 +1,352 @@
+              useUnifiedTopology: true,
+            },
+          );
+        });
+
+        it('should add +srv and handle ssl parameter', async () => {
+          expect.assertions(1);
+
+          const database = setupDatabase(context);
+
+          const options = {
+            mongodbSrv: true,
+            dbUser: 'forest',
+            dbPassword: 'forest',
+            dbHostname: 'mongodbhost',
+            dbName: 'mydatabase',
+          };
+          await database.connectToMongodb(options, true);
+
+          expect(context.mongodb.MongoClient.connect).toHaveBeenCalledWith(
+            'mongodb+srv://forest:forest@mongodbhost/mydatabase',
+            {
+              ssl: true,
+              useNewUrlParser: true,
+              useUnifiedTopology: true,
+            },
+          );
+        });
+      });
+
+      describe('when a connectionUrl is provided', () => {
+        it('should call mongodb connect with the specified url', async () => {
+          expect.assertions(1);
+
+          const database = setupDatabase(context);
+
+          const options = {
+            dbConnectionUrl: 'mongodb://forest:forest@mongodbhost:27018/anotherdatabase',
+          };
+          await database.connectToMongodb(options);
+
+          expect(context.mongodb.MongoClient.connect).toHaveBeenCalledWith(
+            'mongodb://forest:forest@mongodbhost:27018/anotherdatabase',
+            {
+              useNewUrlParser: true,
+              useUnifiedTopology: true,
+            },
+          );
+        });
+      });
+    });
+
+    it('should call mongodb db function on successful connection', async () => {
+      expect.assertions(2);
+
+      const mongoDbMock = {
+        db: jest.fn(),
+      };
+      const context = {
+        mongodb: {
+          MongoClient: {
+            connect: jest.fn().mockReturnValue(
+              Promise.resolve(mongoDbMock),
+            ),
+          },
+        },
+      };
+      const database = setupDatabase(context);
+
+      await database.connectToMongodb({ dbName: 'fake' });
+
+      expect(mongoDbMock.db).toHaveBeenCalledTimes(1);
+      expect(mongoDbMock.db).toHaveBeenCalledWith('fake');
+    });
+
+    it('should call handleAuthenticationError on a failed connection', async () => {
+      expect.assertions(2);
+
+      const connectError = new Error('connection error');
+      const context = {
+        mongodb: {
+          MongoClient: {
+            connect: jest.fn().mockReturnValue(Promise.reject(connectError)),
+          },
+        },
+      };
+      const database = setupDatabase(context);
+      const handleAuthenticationErrorSpy = jest
+        .spyOn(database, 'handleAuthenticationError')
+        .mockImplementation(jest.fn);
+
+      await database.connectToMongodb({});
+
+      expect(handleAuthenticationErrorSpy).toHaveBeenCalledTimes(1);
+      expect(handleAuthenticationErrorSpy).toHaveBeenCalledWith(connectError);
+    });
+  });
+
+  describe('connect', () => {
```

Comment:
What about ssl tests? And tests of getDialect call?

Reply:
>  And tests of getDialect call?

You mean in this function, right? Not sure that's mandatory (Since I already check for the createMongoose/sequelize connection)

> ssl tests?

Requires 3 tests to be fullfilled, which is about 75 LOC. Would prefer not to write these, since (to me at least) it indicate that some code should be refactored. I would prefer to refactor the code here in a following PR instead of adding more tests (If that's ok to you).

Reply:
ok for me ;)